docs/asyncapi.yaml

```diff
@@ -209,3 +213,20 @@ components:
                 accountCode:
                   type: string
                   description: Unique account code for budgeting, optional
+    PaymentRequestResponse:
+      contentType: application/json
+      payload:
+        type: object
+        properties:
+          paymentRequest:
+            type: object
+            description: Submitted payment request with enriched values
+            properties:
+              $ref: '#/components/messages/EnrichedPaymentRequest'
+          accepted:
+            type: boolean
+            description: Whether the payment request was accepted by the payment service
+          error:
+            type: string
+            description: Only present if payment request was rejected detailing hte reason why
+
```

Comment:
Commenting here as lines are not modified.

Fields which have specific formats required which are not mentioned in doc:

- `deliveryBody`: regex pattern not stated
- `ledger`': mentions "and" when it needs clearer it's AP or AR
- `frn`: is required but doc states, optional if `sbi` is given -- not logically implemented (also its valid numeric range is not given)
- `sbi`: similar to `frn` above, except its optional regardless (no numeric range specified here either)
- `marketingYear`: numeric range not given
- `currency`: only once defines if not given it defaults to "GBP", doesn't mention only it and "EUR" valid
- `schedule`: mention other 2 valid options
- `dueDate`: only once defines if not given it defaults to current date
- `originalSettlementDate`: no mention in here
- `correlationId`: mention its a v4 `guid`

Reply:
I've re-written this AsyncAPI doc as it wasn't written in an overly manageable way.  I've added reusable schemas for the data types so it can be updated easier.

Added examples, requirements and constraints using correct spec.